dev/functional/tests/cypress/page-objects/student-objects/ApplicationObject.ts

```diff
@@ -339,55 +336,55 @@ export default class ApplicationObject {
   }
 
   iConfirmMyStudentAidCheckbox() {
-    return cy.get("[name='data[studentInfoConfirmed]']");
+    return cy.contains("I confirm my StudentAid");
   }
 
   citizenStatusRadioButton() {
     return cy.contains("Canadian citizen");
   }
 
   residentOfBCRadioButton() {
-    return cy.get("#e8a3hql-no");
+    return cy.get("[role='radio']").eq(4);
```

Comment:
Getting the element based on the index is way too weak and subject to be broken at any UI change. Is it a kind of temporary solution?

Reply:
Not sure why, but specific places this problem occurs.
As you can see from below image, there are few things in common with selectors.
a) id
b) name
<img width="592" alt="issue4" src="https://user-images.githubusercontent.com/98920571/170556151-31410809-3018-4994-b72f-a0590b8c2c5a.png">

**(a) id**
id changes every time automatically, so selectors also change every time.
<img width="689" alt="issue2" src="https://user-images.githubusercontent.com/98920571/170557060-d4cb7668-978d-4e3c-ae56-4d81ddfbc70c.png">
<img width="691" alt="issue1" src="https://user-images.githubusercontent.com/98920571/170557062-552b7914-207b-4155-8a19-37da22ed5fe7.png">

**(b) name**
Test fails because the name selector cannot be found.
<img width="940" alt="issue3" src="https://user-images.githubusercontent.com/98920571/170557592-d64d2358-0271-44a4-ae91-0303dc715401.png">

Reply:
Please schedule a meeting to go over this.